### PR TITLE
--Setup dataset test data for MetadataMediator Implementations.

### PR DESCRIPTION
**Pull request by jturner65**
## Motivation and Context
Setup initial dataset test data structure for Metadata Mediator testing process.  The MM process is predicated on loading a dataset that is described by specific configuration files which will be assumed to be in some kind of hierarchy.  This test dataset provides this hierarchy, isolated from other test assets so that any modifications to the common test assets will not impact the dataset tests.

These configuration files follow the format specified [here](https://drive.google.com/drive/folders/18FtczpVwbe3okYwiofVCbsX1cawFCUN7) 

## How Has This Been Tested
c++ and python

## Types of changes

- [x] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: data/test_assets/dataset_tests/test_dataset.dataset_config.json [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        "default_attributes": {
+        },
+        "paths": ["lights"],
+        "configs" : [
+        {
+            "original_file":"dataset_test_lights.lighting_config.json",
+            "template_handle": "modified_test_lights",
+            "attributes": {
+  			    "lights":{
+					"0": { "position": [1.5,0.1,1.5], "intensity": 2.4, "color": [0.5,1,0.95], "type": "point"},
```
Comment (bigbike) [inline data/test_assets/dataset_tests/test_dataset.dataset_config.json:71]: May I ask what the relation is between the lights setup here and the lights specified in the  dataset_tests/lights/dataset_test_lights.lighting_config.json?

Comment (jturner65) [reply data/test_assets/dataset_tests/test_dataset.dataset_config.json:71]: This is a way for an individual to create a new lighting setup ("modified_test_lights"), based on an existing one ("dataset_test_lights.lighting_config.json"), where specified lighting values are overridden - in this case, lights with IDs 0,1 and 11.

File: data/test_assets/dataset_tests/lights/dataset_test_lights.lighting_config.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    "lights":{
+        "0": { "position": [2.45833,0.1,3.84271], "intensity": 1.4, "color": [1,1,0.95], "type": "point"},
+        "1": { "position": [3.28035,-0.1,8.15357], "intensity": 1.1, "color": [1,1,0.95], "type": "point"},
+        "2": { "position": [0.35,0.5,3.5], "intensity": 1.3, "color": [0.93,0.98,1], "type": "point"},
+        "3": { "position": [0.35,0.5,1.0], "intensity": 1.3, "color": [0.93,0.98,1], "type": "point"},
+        "4": { "direction": [0.0,-1.0,0.0], "intensity": 1.3, "color": [0.93,0.98,1], "type": "area"},
+        "5": { "position": [-5.59517,5,-8.33407], "intensity": 120, "color": [1,1,1], "type": "point"},
+        "6": { "position": [8.65492,1.7,3.23571], "intensity": 15, "color": [1,1,1], "type": "point"},
+        "7": { "position": [5,5,0], "intensity": -10, "color": [1,1,1], "type": "point"},
+        "8": { "direction": [1.80922,-1.0,0.5], "intensity": -0.4, "color": [1,1,1], "type": "area"},
```
Comment (eundersander) [inline data/test_assets/dataset_tests/lights/dataset_test_lights.lighting_config.json:11]: I don't think we support area lights. And wouldn't an area light need to specify the area somehow?

Also what happened to directional lights?

Comment (jturner65) [reply data/test_assets/dataset_tests/lights/dataset_test_lights.lighting_config.json:11]: Thanks Eric.  That's a typo, should be directional.

Comment (Skylion007) [reply data/test_assets/dataset_tests/lights/dataset_test_lights.lighting_config.json]: @jturner65 Do not have any validators to catch these errors?

Comment (jturner65) [reply data/test_assets/dataset_tests/lights/dataset_test_lights.lighting_config.json]: Not yet, not in this PR.  This is just the initial commit of the testing data used to verify the functionality of the dataset and scene import.  JSON validators would be great, however.

Comment (Skylion007) [reply data/test_assets/dataset_tests/lights/dataset_test_lights.lighting_config.json]: @jturner65 Done: https://github.com/facebookresearch/habitat-sim/pull/854

Comment (jturner65) [reply data/test_assets/dataset_tests/lights/dataset_test_lights.lighting_config.json]: Just FYI, in this case, the issue was incorrect data value for "type", not any invalid json configurations.   The added validator does not check for values, does it?

File: data/test_assets/dataset_tests/test_dataset.dataset_config.json [hunk 3]
```diff
@@ -0,0 +1,102 @@
+{
+    "stages": {
+        "default_attributes": {
+            "origin":[1.0,2.0,3.0]
+        },
+        "paths": ["stages"],
+        "configs" : [
+        {
+            "original_file": "dataset_test_stage.stage_config.json",
```
[the remark this answers is not in the document]
Comment (jturner65) [reply data/test_assets/dataset_tests/test_dataset.dataset_config.json:9]: Yes, you are correct, this is what we discussed.

**Review by aclegg3 — APPROVED** [review]
👍

**Pull request merged**